### PR TITLE
REPO-3329 record deployment method

### PR DESCRIPTION
**Pull request by eknizat**
* Add DeploymentMethodProvider which reads the deployment method property "deployment.method"
* Extend InfoDataCollector to include deployent method value
* Set default value for deployment.method property in repository.properties

### CODE REVIEW

File: src/main/java/org/alfresco/heartbeat/InfoDataCollector.java [hunk 1]
```diff
@@ -115,6 +125,10 @@ public void afterPropertiesSet() throws Exception
         infoValues.put("version", version);
         infoValues.put("schema", new Integer(serverDescriptor.getSchema()));
         infoValues.put("edition", serverDescriptor.getEdition());
+        infoValues.put("deploymentMethod", deploymentMethodProvider.getDeploymentMethod().toString());
+
+        // remove this
```
Comment (derekhulley) [inline src/main/java/org/alfresco/heartbeat/InfoDataCollector.java]: I agree.

File: src/main/resources/alfresco/heartbeat/heartbeat-context.xml [hunk 2]
```diff
@@ -20,6 +20,7 @@
         <property name="hbDataSenderService" ref="hbDataSenderService" />
         <property name="scheduler" ref="schedulerFactory" />
         <property name="jobLockService" ref="jobLockService" />
+        <property name="testMode" value="true" />
```
Comment (derekhulley) [inline src/main/resources/alfresco/heartbeat/heartbeat-context.xml]: Is this correct?  Should this be in test mode by default?

File: src/main/resources/alfresco/core-services-context.xml [hunk 3]
```diff
@@ -1352,4 +1352,11 @@
       <property name="policyComponent" ref="policyComponent" />
       <property name="namespaceService" ref="namespaceService" />
    </bean>
+
+   <bean id="deploymentMethodProvider" class="org.alfresco.repo.deployment.DeploymentMethodProviderImpl" >
+      <property name="deploymentMethodAsString">
```
Comment (derekhulley) [inline src/main/resources/alfresco/core-services-context.xml]: I think that Spring will do the conversion.  Have you checked?

File: src/main/resources/alfresco/heartbeat/heartbeat-context.xml (hunk 2 above)
Comment (eknizat) [reply src/main/resources/alfresco/heartbeat/heartbeat-context.xml]: This is just to schedule all collectors at 1min intervals, I will remove it before merge

File: src/main/resources/alfresco/core-services-context.xml (hunk 3 above)
Comment (eknizat) [reply src/main/resources/alfresco/core-services-context.xml]: Yes you are right, Spring does indeed do the conversion. I'll change it in the next commit.

**Comment by derekhulley** [comment]
Could you add a comment on why this MR was closed, please @eknizat.

**Comment by eknizat** [comment]
The MR was closed and its branch deleted and replaced by this MR: https://github.com/Alfresco/alfresco-repository/pull/111
This was done for clarity as the original branch was used to investigate a failing test and had irrelevant commits to the REPO-3329 feature.

**Pull request closed without merging**